wcfsetup/install/files/lib/system/cronjob/RefreshSearchRobotsCronjob.class.php

```diff
@@ -15,15 +16,23 @@
  * @package	WoltLabSuite\Core\System\Cronjob
  */
 class RefreshSearchRobotsCronjob extends AbstractCronjob {
+	/**
+	 * URL to the spider list.
+	 */
+	const SPIDER_LIST_URL = 'http://assets.woltlab.com/spiderlist/typhoon/list.xml';
+	
 	/**
 	 * @inheritDoc
 	 */
 	public function execute(Cronjob $cronjob) {
 		parent::execute($cronjob);
 		
-		$filename = FileUtil::downloadFileFromHttp('http://assets.woltlab.com/spiderlist/typhoon/list.xml', 'spiders');
+		$client = HttpFactory::getDefaultClient();
+		$request = new Request('GET', self::SPIDER_LIST_URL);
+		$response = $client->send($request);
+		
 		$xml = new XML();
-		$xml->load($filename);
+		$xml->loadXML('list.xml', $response->getBody()->getContents());
```

Comment:
```suggestion
		$xml->loadXML('list.xml', (string) $response->getBody());
```
Reason: The PSR-7 API is not well-designed in this regard. `->getContents()` returns the *remaining* contents. This most likely is all of the contents here, but a bit fragile and misleading. `->__toString()` effectively is a `->rewind()` + `->getContents()` and thus is safer.